coderd/coderdtest/coderdtest.go

```diff
@@ -645,10 +646,10 @@ func (c *ProvisionerdCloser) Close() error {
 // well with coderd testing. It registers the "echo" provisioner for
 // quick testing.
 func NewProvisionerDaemon(t testing.TB, coderAPI *coderd.API) io.Closer {
-	return NewTaggedProvisionerDaemon(t, coderAPI, "test", nil)
+	return NewTaggedProvisionerDaemon(t, coderAPI, "test", "", nil)
 }
 
-func NewTaggedProvisionerDaemon(t testing.TB, coderAPI *coderd.API, name string, provisionerTags map[string]string) io.Closer {
+func NewTaggedProvisionerDaemon(t testing.TB, coderAPI *coderd.API, name string, versionOverride string, provisionerTags map[string]string) io.Closer {
```

Comment:
I really dislike this argument. seeing a bunch of calls with some random parameter in the middle that's usually just `""` doesn't feel like the right design, even for a testing function.

Reply:
didn't really address or explain this

Reply:
My fault. 

I will see if I can find an alternate way to inject this. I might just make the arguments a struct instead

Reply:
How do you feel about the option functions pattern?

`coderd.MemoryProvisionerWithVersionOverride(options.ProvisionerDaemonVersion)`

https://github.com/coder/coder/blob/df65b926a09b6c4e800a4261209fea5daa70e550/coderd/coderdtest/coderdtest.go#L605-L605

So the existing calls are untouched

Reply:
that would be great!